crates/rpc/src/handlers/light_client.rs

```diff
@@ -161,3 +161,133 @@ pub async fn get_light_client_updates(
+            ApiError::InternalError(format!(
+                "Failed to get latest slot, error: {err:?}"
+            ))
+        })?
+        .ok_or_else(|| {
+            ApiError::NotFound("LC finality update unavailable".into())
+        })?;
+
+    let head_block_root = db
+        .slot_index_provider()
+        .get(latest_slot)
+        .map_err(|err| {
+            ApiError::InternalError(format!(
+                "Failed to get block root for latest slot, error: {err:?}"
+            ))
+        })?
+        .ok_or_else(|| {
+            ApiError::NotFound("LC finality update unavailable".into())
+        })?;
+
+    // Get the head block and state
+    let head_block = db
+        .beacon_block_provider()
+        .get(head_block_root)
+        .map_err(|err| {
+            ApiError::InternalError(format!(
+                "Failed to get head block, error: {err:?}"
+            ))
+        })?
+        .ok_or_else(|| {
+            ApiError::NotFound("LC finality update unavailable".into())
+        })?;
+
+    let head_state = db
+        .beacon_state_provider()
+        .get(head_block_root)
+        .map_err(|err| {
+            ApiError::InternalError(format!(
+                "Failed to get head state, error: {err:?}"
+            ))
+        })?
+        .ok_or_else(|| {
+            ApiError::NotFound("LC finality update unavailable".into())
+        })?;
+
+    // Get the finalized block and state
+    let finalized_block = db
+        .beacon_block_provider()
+        .get(finalized_checkpoint.root)
+        .map_err(|err| {
+            ApiError::InternalError(format!(
+                "Failed to get finalized block, error: {err:?}"
+            ))
+        })?
+        .ok_or_else(|| {
+            ApiError::NotFound("LC finality update unavailable".into())
+        })?;
+
+    let finalized_state = db
+        .beacon_state_provider()
+        .get(finalized_checkpoint.root)
+        .map_err(|err| {
+            ApiError::InternalError(format!(
+                "Failed to get finalized state, error: {err:?}"
+            ))
+        })?
+        .ok_or_else(|| {
+            ApiError::NotFound("LC finality update unavailable".into())
+        })?;
+
+    // Create the finality update
+    let finality_update = LightClientFinalityUpdate {
+        attested_header: head_state.latest_block_header.clone(),
+        finalized_header: finalized_state.latest_block_header.clone(),
+        finality_branch: head_state.finalized_root_inclusion_proof().map_err(|err| {
+            ApiError::InternalError(format!(
+                "Failed to get finalized root inclusion proof, error: {err:?}"
+            ))
+        })?.into(),
+        sync_aggregate: head_block.message.body.sync_aggregate,
+        signature_slot: head_block.message.slot,
+    };
+
+    // Check Accept header for response format
+    let accept = req.headers().get("accept").and_then(|h| h.to_str().ok());
+    let response = match accept {
```

Comment:
```suggestion
    // Check Accept header for response format
    let response = match http_request.headers().get("accept").and_then(|header| header.to_str().ok()) {
```

we might as well inline this